tests/test_rag.py

```
@@ -0,0 +1,111 @@
+
+from langchain_core.documents import Document
+from langchain_core.language_models.fake import FakeListLLM
+from langchain_core.messages import HumanMessage, AIMessage
+
+from rag import format_docs, contextualize_query, stream_answer, _merge_results, load_resources
+
+
+# ── load_resources ────────────────────────────────────────
+
+def test_load_resources_requires_allow_deserialization():
+    """allow_deserialization=False（デフォルト）では ValueError を送出する。"""
```

Ruff (0.15.7)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)